### ISSUE TITLE
[Request] At least partial support of native windows openssh

### ISSUE THREAD

**Issue by eugenov** [billziss-gh/sshfs-win#323]
Hello!
What do you think about implementation of openssh support, such as:
- reading/parcing ~\.ssh\config
- integration with ssh-agent.exe
For seamless ssh-ing it would be enough. Main goal is creating environment, close to linux file managers (ssh bookmarks, shortcuts, etc.), but on Win.
Thank you!

**Comment by bbutler-tssg** [comment]
I agree that supporting ssh config files would improve greatly usability. Indeed, potentially the same ssh config file could be used across Windows, linux and macos clients, because the ssh config format is well able to express the details of the ssh connection with the server.

I also get the impression that sshfs-win is limited to id_rsa keys - it would be great to support id_ed25519 keys as well. At least I get an ssh "connection reset by peer" when I try to use such keys.

Lastly, it is considered good practice to add password checking to the private key, and on unix boxes to use ssh-agent to store the password for each password-protected private key. ssh-agent can be found in both Git for Windows and WSL (e.g., for logging in to git repos), but sshfs-win does not seem to use it.

**Comment by billziss-gh** [comment]
A few years ago I looked into supporting OpenSSH.

While I do not remember all the details I had determined that it was not easy to support. If I recall correctly the issue was that SSHFS communicates with SSH using the Cygwin implementation of UNIX sockets, which was incompatible with anything that OpenSSH provided.

I understand that this issue is asking for partial integration with OpenSSH (via configuration sharing). Unfortunately I do not have the time to work on anything like this myself, but if someone wants to take it on I am happy to provide support.

**Comment by bbutler-tssg** [comment]
Thanks @billziss-gh - you are right, my request mostly related to shared configuration, but also to greater support for OpenSSH features, which might be a step too far at this time, especially if sockets are handled differently.
